services/main.services.js

```diff
@@ -0,0 +1,84 @@
+const httpStatus = require("http-status");
+const { Product } = require("../models");
+const ApiError = require("../utils/ApiError.js");
+const { handleMessage } = require("../utils/handleMessage");
+const { queryHandler } = require("../utils/queryHandler");
+const { getData } = require("../utils/getData");
+const { sendEmail } = require("../utils/sendEmail");
+
+/**
+ * Verification
+ * @param {Object} data
+ * @returns {Promise<User>}
+ */
+const getMain = (data) => {
+  const VERIFY_TOKEN = process.env.VERIFY_TOKEN;
+  // Parse the query params
+  const mode = data.query["hub.mode"];
+  const token = data.query["hub.verify_token"];
+  const challenge = data.query["hub.challenge"];
+
+  // Checks if a token and mode is in the query string of the request
+  if (mode && token) {
+    if (mode === "subscribe" && token === VERIFY_TOKEN) {
+      return challenge;
+    }
+  } else {
+    throw new ApiError(
+      httpStatus.BAD_REQUEST,
+      "Invalid webhook verification token"
+    );
+  }
+};
+
+/**
+ * Listening to events
+ * @param {Object} data
+ * @returns {Promise<User>}
+ */
+const postMain = (data) => {
+  let values;
+  if (data.body.object === "page") {
+    data.body.entry.forEach((entry) => {
+      // Gets the body of the webhook event
+      let webhook_event = entry.messaging[0];
+      // Get the sender PSID
+      let sender_psid = webhook_event.sender.id;
+      if (webhook_event.message) {
+        values = [sender_psid, webhook_event];
+      }
+    });
+    return values;
+  }
+};
+
+const postHandler = async (sender_psid, data) => {
+  // Check type of query
+  const query = data.message.text.split(" ")[0];
+  const productQuery = data.message.text.split(" ")[1];
+  // Send to query handler
+  const queryType = queryHandler(query);
+  // Get best answer for the query
+  if (queryType === "infoRequest") {
+    // handle query for info request
+    const message = await getData(query, data);
+    handleMessage(sender_psid, data, message);
+  } else if (queryType === "buyRequest") {
+    // Get product details from database
+    const product = await Product.findOne({ _id: productQuery });
+    // send email to my account for buy request
+    sendEmail(product); 
```

Comment:
Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/ParmenidesSartre/Respond-Task1/pullRequest?prid=9934275)
```suggestion
    sendEmail(product);
```